cfme/fixtures/utility_machine.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+import os.path
+
+import fauxfactory
+import pytest
+
+from cfme.utils.conf import cfme_data
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log import logger
+from cfme.utils.net import find_pingable
+from cfme.utils.virtual_machines import deploy_template
+from cfme.utils.wait import TimedOutError
+from cfme.utils.wait import wait_for
+
+
+@pytest.fixture(scope='module')
+def utility_machine():
+    """ Deploy an utility vm for tests to use.
+
+    This fixture creates a vm on provider and then receives its ip.
+    After the test run vm is deleted from provider.
+    """
+    try:
+        root_password = fauxfactory.gen_alphanumeric(length=10)
+        try:
+            with open(os.path.expanduser('~/.ssh/id_rsa.pub')) as f:
+                authorized_ssh_keys = f.read()
+        except FileNotFoundError:
+            authorized_ssh_keys = None
+        data = cfme_data.utility_machine
+        vm = deploy_template(
+            data.provider,
+            random_vm_name('proxy'),
+            template_name=data.template_name,
+            root_password=root_password,
+            authorized_ssh_keys=authorized_ssh_keys
+        )
+    except AttributeError:
+        msg = 'Missing data in cfme_data.yaml, cannot deploy proxy'
+        logger.exception(msg)
+        pytest.skip(msg)
+
+    try:
```

Comment:
Well the concept of waiting for ping seems to me as an "anti-pattern". We should be re-trying the connect() rather then pinging the machine. When the machine is starting-up, it will be pingable but there is some time window when the required service doesn't have the socket created an listening, so to not risk any race, we in the end still need to have a repeated attempts to  connect anyway.